### PR TITLE
fix: remove "allow-privileged" in kubelet 1.15.0

### PR DESCRIPTION
**Pull request by andyzhangx**
**Reason for Change**:

This PR remove `--allow-privileged` in kubelet config from k8s v1.15.0, since `--allow-privileged` is removed in this k8s upstream PR: https://github.com/kubernetes/kubernetes/pull/77820

**Issue Fixed**:

Fixes#1368

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by andyzhangx** [comment]
cc @feiskyer

File: pkg/api/defaults-kubelet.go [hunk 1]
```diff
@@ -186,6 +186,13 @@ func removeKubeletFlags(k map[string]string, v string) {
 		}
 	}
 
+	// Get rid of values not supported in v1.15 and up
+	if common.IsKubernetesVersionGe(v, "1.15.0-alpha.3") {
```
[the remark this answers is not in the document]
Comment (feiskyer) [reply pkg/api/defaults-kubelet.go]: The PR https://github.com/kubernetes/kubernetes/pull/77820 is merged 6 days ago, and hasn't been included in any releases yet. That means > 1.15.0-beta.0.

[the remark this answers is not in the document]
Comment (CecileRobertMichon) [reply pkg/api/defaults-kubelet.go]: so in that case should it be `if common.IsKubernetesVersionGe(v, "1.15.0-beta-0") {` ?

**Comment by andyzhangx** [comment]
/azp run pr-e2e
sth wrong with github...

**Comment by mboersma** [comment]
We should add a test case for this to defaults-kubelet_test.go.

**Comment by feiskyer** [comment]
@andyzhangx Could you add a unit test for this?

@mboersma Could you help to cut a release after this gets merged?

**Comment by mboersma** [comment]
@andyzhangx I added a unit test just now that should cover the flag removal. I'll ask the @Azure/aks-engine team about a release.

**Comment by CecileRobertMichon** [comment]
@feiskyer why is this needed in a patch release? 1.15 is still in pre-release and -beta.1 support wasn't released yet.

**Comment by feiskyer** [comment]
@CecileRobertMichon it's required for upstream pull-requests CI

**Pull request merged**